hooks/pre-commit

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+from __future__ import print_function
+from future import standard_library
+standard_library.install_aliases()
+import os
+import urllib.request
+import urllib.parse
+import urllib.error
+from distutils.version import LooseVersion
+import re
+import sys
+import subprocess
+
+pattern = re.compile(r'[."]version["\s":=]+(\d+\.\d+.\d+)')
+version_dict = {}
+version_file = urllib.request.urlopen(
+    "https://raw.githubusercontent.com/weecology/retriever/master/version.txt")
+version_file.readline()
+
+
+def to_string(value):
+    if sys.version_info >= (3, 0, 0):
+        return value.decode("UTF-8")
+    else:
+        return value
+
+
+for lines in version_file:
+    key_values = to_string(lines).split(",")
+    version_dict[key_values[0]] = key_values[1]
+
+
+def get_script_version(file_name):
+    file_name = os.path.split(file_name)[1]
+    return LooseVersion(version_dict[file_name])
+
+
+# check if upstream is added as remote
+remotes = to_string(subprocess.check_output("git remote -v", shell=True))
+
+if u"upstream\thttps://github.com/weecology/retriever.git (fetch)" not in remotes:
+    print("\nPlease first add upstream as a remote."
+          "\ngit remote add upstream https://github.com/weecology/retriever.git")
+    exit(1)
+
+# check if repo is up-to-date with upstream master
+logs_update = to_string(
+    subprocess.check_output("git fetch upstream", shell=True))
+logs_ahead = to_string(
+    subprocess.check_output(
+        "git log --oneline upstream/master..HEAD | wc -l", shell=True))
+logs_behind = to_string(
+    subprocess.check_output(
+        "git log --oneline HEAD..upstream/master | wc -l", shell=True))
+if int(logs_ahead) > 0 and int(logs_behind) > 0:
+    print(
+        "\nCurrent branch is behind upstream/master."
+        "\nPlease update your branch with upstream master before making commits")
+    exit(1)
```

Comment:
Thanks for the clarification @zhangcandrew. If the script itself has been changed in `upstream/master` then I think we do need to rebase since by definition the version lines would run into conflicts. That said, I don't think this will happen very often and it will show up once the PR is submitted. So, it seems like we have two choices: 1) assume this won't happen that often and deal with it in review; or 2) check to see if there are upstream changes to a changed script.

If we want to try to go for (2) I think we'd want to check the last commit hash from upstream master for the changed script using either:

```
git log --oneline upstream/master scripts/test.json
```

or some version of `git rev-parse` I can't figure out. Then see if that hash is present in the current branch or not. If so, good to go, if not, rebase.

Since @henrykironde is hoping to get this in today I'd suggest we go with (1) and open an issue to improve things by adding (2).